src/App.tsx

```diff
@@ -16,60 +16,150 @@ const goodsFromServer = [
 ];
 
 enum SortType {
-  NONE,
-  ALPABET,
-  LENGTH,
+  NONE = 'NONE',
+  ALPABET = 'ALPABET',
+  LENGTH = 'LENGTH',
+  REVERSE = 'REVERSE',
 }
 
-// Use this function in the render method
-function getReorderedGoods(
-  goods: string[],
-  sortType: SortType,
-  isReversed: boolean,
-) {
-  // Not to mutate the original array
-  const visibleGoods = [...goods];
-
-  // Sort and reverse goods if needed
-  // ...
-
-  return visibleGoods;
-}
-
-// DON'T save goods to the state
 type State = {
   isStarted: boolean,
-  isReversed: boolean,
-  sortType: SortType,
+  visibleGoods: string[],
 };
 
-export const App = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+export class App extends React.Component<{}, State> {
+  state: State = {
+    isStarted: false,
+    visibleGoods: goodsFromServer,
+  };
+
+  sortGoods = (sortBy: SortType) => {
+    this.setState(state => {
+      const newGoods = [...state.visibleGoods];
+
+      switch (sortBy) {
+        case SortType.ALPABET:
+          newGoods.sort((good1, good2) => good1.localeCompare(good2));
+          break;
+
+        case SortType.LENGTH:
+          newGoods.sort((good1, good2) => good1.length - good2.length);
+          break;
+
+        case SortType.REVERSE:
+          newGoods.reverse();
+          break;
+
+        default:
+          break;
+      }
+
+      return {
+        visibleGoods: newGoods,
+      };
+    });
+  };
+
+  reset = () => {
+    this.setState({ visibleGoods: goodsFromServer });
+  };
+
+  startListView = () => {
+    const { isStarted } = this.state;
+
+    if (!isStarted) {
+      this.setState(() => ({ isStarted: !isStarted }));
+    }
```

Comment:
just use functional setState cllback params
```suggestion
    this.setState((state) => ({ isStarted: !state.isStarted }));
```